ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+                    version = get(user, 'version')
+                    user_dict = {
+                        'name': get(user, 'name'),
+                        'group': get(user, 'group'),
+                        'version': version
+                    }
+                    compute_v3_user_localized_key = get(snmp_settings, 'compute_v3_user_localized_key')
+                    if (
+                            version == 'v3' and
+                            compute_local_engineid is True and
+                            compute_v3_user_localized_key is True
+                    ):
+                        user_dict.update({'localized': local_engine_id})
+
+                    auth = get(user, 'auth')
+                    auth_passphrase = get(user, 'auth_passphrase')
+                    if (
+                            version == 'v3' and
+                            auth is not None and
+                            auth_passphrase is not None
+                    ):
+                        user_dict.update({'auth': auth})
+                        hash_filter = {}
+                        if (
+                                compute_local_engineid is True and
+                                compute_v3_user_localized_key is True
+                        ):
+                            hash_filter.update({"passphrase": auth_passphrase, "auth": auth,
+                                                "engine_id": local_engine_id})
+                            user_dict.update({'auth_passphrase': self._hash_passphrase(hash_filter)})
+                        else:
+                            user_dict.update({'auth_passphrase': auth_passphrase})
+
+                        priv = get(user, 'priv')
+                        priv_passphrase = get(user, 'priv_passphrase')
+                        if (
+                            priv is not None and
+                            priv_passphrase is not None
+                        ):
+                            user_dict.update({'priv': priv})
+                            if (
+                                    compute_local_engineid is True and
+                                    compute_v3_user_localized_key is True
+                            ):
+                                hash_filter.update({"passphrase": priv_passphrase, "priv": priv})
+                                user_dict.update({'priv_passphrase': self._hash_passphrase(hash_filter)})
+                            else:
+                                user_dict.update({'priv_passphrase': priv_passphrase})
+                    snmp_server['users'].append(user_dict)
+            return snmp_server
+        return None
+
+    @cached_property
+    def spanning_tree(self):
+        spanning_tree_root_super = get(self._switch_data, 'spanning_tree_root_super')
+        spanning_tree_mode = get(self._switch_data, 'spanning_tree_mode')
+        if spanning_tree_root_super is True or spanning_tree_mode is not None:
```

Comment:
as usual let's prefer a syntax like

```suggestion
        if spanning_tree_root_super is False and spanning_tree_mode is None:
            return None
        ...
```